crates/signer/src/manager/dirk.rs

```diff
@@ -0,0 +1,917 @@
+use std::{
+    collections::{HashMap, HashSet},
+    fmt::Debug,
+    fs,
+    path::PathBuf,
+};
+
+use alloy::{hex, primitives::FixedBytes, rpc::types::beacon::constants::BLS_SIGNATURE_BYTES_LEN};
+use blsful::inner_types::{Field, G2Affine, G2Projective, Group, Scalar};
+use cb_common::{
+    commit::request::{ConsensusProxyMap, ProxyDelegation, SignedProxyDelegation},
+    config::DirkConfig,
+    constants::COMMIT_BOOST_DOMAIN,
+    signature::compute_domain,
+    signer::{BlsPublicKey, BlsSignature, ProxyStore},
+    types::{Chain, ModuleId},
+};
+use rand::Rng;
+use tonic::transport::{Channel, ClientTlsConfig};
+use tracing::{info, trace, warn};
+use tree_hash::TreeHash;
+
+use crate::{
+    error::SignerModuleError::{self, DirkCommunicationError},
+    proto::v1::{
+        account_manager_client::AccountManagerClient, lister_client::ListerClient,
+        sign_request::Id as SignerId, signer_client::SignerClient, Account as DirkAccount,
+        DistributedAccount, GenerateRequest, ListAccountsRequest, ResponseState, SignRequest,
+        UnlockAccountRequest,
+    },
+};
+
+#[derive(Clone, Debug)]
+enum WalletType {
+    Simple,
+    Distributed,
+}
+
+#[derive(Clone, Debug)]
+struct HostInfo {
+    domain: String,
+    participant_id: u64,
+}
+
+#[derive(Clone, Debug)]
+struct Account {
+    wallet: String,
+    name: String,
+    public_key: Option<BlsPublicKey>,
+    hosts: Vec<HostInfo>,
+    wallet_type: WalletType,
+    signing_threshold: u32,
+    is_proxy: bool,
+}
+
+impl Account {
+    pub fn complete_name(&self) -> String {
+        format!("{}/{}", self.wallet, self.name)
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct DirkManager {
+    chain: Chain,
+    channels: HashMap<String, Channel>, // domain -> channel
+    accounts: HashMap<String, Account>, // pubkey -> account
+    unlock: bool,
+    secrets_path: PathBuf,
+    proxy_store: Option<ProxyStore>,
+}
+
+impl DirkManager {
+    pub async fn new_from_config(chain: Chain, config: DirkConfig) -> eyre::Result<Self> {
+        let mut tls_configs = Vec::new();
+
+        // Create a TLS config for each host
+        for host in config.hosts.clone() {
+            let mut tls_config = ClientTlsConfig::new().identity(config.client_cert.clone());
+
+            if let Some(ca) = &config.cert_auth {
+                tls_config = tls_config.ca_certificate(ca.clone());
+            } else {
+                trace!(?host.domain, "CA certificate for domain not found");
+            }
+
+            tls_config = tls_config.domain_name(host.domain.unwrap_or_default());
```

Comment:
We should not use `unwrap_or_default()` here. In case `domain` is `None` the function shouldn't be called.

Reply:
Done in 46033ea281cf1d9570c1c8a6dc701e95fff0f2e7